cmd/geth/admin.go

```diff
@@ -396,28 +400,243 @@ func (js *jsre) printBlock(call otto.FunctionCall) otto.Value {
 }
 
 func (js *jsre) dumpBlock(call otto.FunctionCall) otto.Value {
-	var block *types.Block
-	if len(call.ArgumentList) > 0 {
-		if call.Argument(0).IsNumber() {
-			num, _ := call.Argument(0).ToInteger()
-			block = js.ethereum.ChainManager().GetBlockByNumber(uint64(num))
-		} else if call.Argument(0).IsString() {
-			hash, _ := call.Argument(0).ToString()
-			block = js.ethereum.ChainManager().GetBlock(common.HexToHash(hash))
-		} else {
-			fmt.Println("invalid argument for dump. Either hex string or number")
-		}
-
-	} else {
-		block = js.ethereum.ChainManager().CurrentBlock()
-	}
-	if block == nil {
-		fmt.Println("block not found")
+	block, err := js.getBlock(call)
+	if err != nil {
+		fmt.Println(err)
 		return otto.UndefinedValue()
 	}
 
 	statedb := state.New(block.Root(), js.ethereum.StateDb())
 	dump := statedb.RawDump()
 	return js.re.ToVal(dump)
+}
 
+func (js *jsre) waitForBlocks(call otto.FunctionCall) otto.Value {
```

Comment:
waits until block height is minimum the number given (guarded by timeout)
basically a sleep function on variable block units of time using a subscription to the chains  head. 
- can be used like (wait for x confirmations), without filters (cos of forks, the resulting state needs to double checked)
- if wrapped around a mining session. It can be used to advance the blockchain in a piecemeal fashion on demand. good for debugging transaction inclusions on a private network, see this script: 
https://gist.github.com/zelig/21ac8eada617e56866d8
- could be used in contract integration tests 

[need to fix js argument handling]